File: services/python-worker/create_org.py
#!/usr/bin/env python

import sys

from utils import get_neo4j_driver
neo4j_driver = get_neo4j_driver()

if len(sys.argv) <= 1:
  print("Usage: create-org NAME")
  exit()

name = sys.argv[1]

print(f"Creating org: {name}")

session = neo4j_driver.session()
tx = session.begin_transaction()
tx.run('''
CREATE (:Organization { name: $name })
''', name=name)
tx.commit()

File: server/tasks/create-org.ts
import { safeQuery } from '../neo4j/connection'

if (process.argv.length !== 3) {
  console.log('Usage: create-org NAME')
  process.exit(1)
}

const name = process.argv[2]

console.log(`Creating org: ${name}`)

safeQuery('CREATE (:Organization { name: $name })', { name })
  .then(() => process.exit(0))
  .catch(() => process.exit(1))
